x-pack/plugins/cases/public/components/case_view/case_view_page.tsx

```diff
@@ -0,0 +1,447 @@
+          if (descriptionUpdate.length > 0) {
+            callUpdate('description', descriptionUpdate);
+          }
+          break;
+        case 'tags':
+          const tagsUpdate = getTypedPayload<string[]>(value);
+          callUpdate('tags', tagsUpdate);
+          break;
+        case 'status':
+          const statusUpdate = getTypedPayload<CaseStatuses>(value);
+          if (caseData.status !== value) {
+            callUpdate('status', statusUpdate);
+          }
+          break;
+        case 'settings':
+          const settingsUpdate = getTypedPayload<CaseAttributes['settings']>(value);
+          if (caseData.settings !== value) {
+            callUpdate('settings', settingsUpdate);
+          }
+          break;
+        default:
+          return null;
+      }
+    },
+    [updateCaseProperty, handleUpdateField, caseData]
+  );
+  return { onUpdateField, isLoading, loadingKey };
+};
+
+export const CaseViewPage = React.memo<CaseViewPageProps>(
+  ({
+    caseData,
+    caseId,
+    fetchCase,
+    onComponentInitialized,
+    actionsNavigation,
+    ruleDetailsNavigation,
+    showAlertDetails,
+    subCaseId,
+    updateCase,
+    useFetchAlertData,
+    refreshRef,
+  }) => {
+    const { userCanCrud } = useCasesContext();
+    const { metricsFeatures } = useCasesFeatures();
+    const { getCaseViewUrl } = useCaseViewNavigation();
+    useCasesTitleBreadcrumbs(caseData.title);
+
+    const [initLoadingData, setInitLoadingData] = useState(true);
+    const init = useRef(true);
+    const timelineUi = useTimelineContext()?.ui;
+
+    const {
+      caseUserActions,
+      fetchCaseUserActions,
+      caseServices,
+      hasDataToPush,
+      isLoading: isLoadingUserActions,
+      participants,
+    } = useGetCaseUserActions(caseId, caseData.connector.id, subCaseId);
+
+    const refetchCaseUserActions = useCallback(() => {
+      fetchCaseUserActions(caseId, caseData.connector.id, subCaseId);
+    }, [caseId, fetchCaseUserActions, subCaseId, caseData]);
+
+    const {
+      metrics,
+      isLoading: isLoadingMetrics,
+      fetchCaseMetrics,
+    } = useGetCaseMetrics(caseId, metricsFeatures);
+
+    const handleRefresh = useCallback(() => {
+      fetchCase();
+      fetchCaseMetrics();
+      refetchCaseUserActions();
+    }, [fetchCase, refetchCaseUserActions, fetchCaseMetrics]);
+
+    const handleUpdateCase = useCallback(
+      (newCase: Case) => {
+        updateCase(newCase);
+        fetchCaseUserActions(caseId, newCase.connector.id, subCaseId);
+        fetchCaseMetrics();
+      },
+      [updateCase, fetchCaseUserActions, caseId, subCaseId, fetchCaseMetrics]
+    );
+
+    const handleUpdateField = useCallback(
+      (newCase: Case, updateKey: UpdateKey) => {
+        updateCase({ ...newCase, comments: caseData.comments });
+        fetchCaseUserActions(caseId, newCase.connector.id, subCaseId);
+        if (updateKey === 'connector' || updateKey === 'status') {
```

Comment:
Yes, I agree with @jonathan! I am gonna forget also for sure :)

Reply:
Yes, I forgot to talk with you about how granular we wanted to be here, I wrote this line just as a reminder. The reason is that this request will not be cheap, it calculates the metrics for all the KPIs, it may become slower as we continue adding metrics calculations. 
Currently, it will be requested every time the user changes anything (the title, tags...). My idea was to parametrize this `updateKey`s into a constant or something centralized, so we have a single definition of the Case changes that we consider require a metrics refresh. 
Even though it makes sense to me to remove the condition, we can also add this optimization later on if needed.

Reply:
Ideally, we would like the updated field to provide that information. All fields should implement an interface and every time we add a new field TS will warn us that the field should implement the interface. For example:

```
interface Fields {
 fetchMetrics: boolean;
}

const statusField: Fields = { fetchMetrics: false }

if (fields[updateKey].fetchMetrics) { ... }
```

Reply:
I like it, this could be a thing to keep in mind the day we refactor the fields logic